lib/driver.js

```diff
@@ -240,13 +240,20 @@ class WebDriverAgentDriver extends BaseDriver {
            !_.includes(commandsToNotProxyToSafari, command);
   }
 
-  async executeCommand (cmd) {
+  async executeCommand (cmd, ...args) {
     if (this.shouldProxyToSafari(cmd)) {
       let safariDriver = await this.getSafariDriver();
-      return await safariDriver.executeCommand(...arguments);
+      return await safariDriver.executeCommand(...args);
     }
+    //check to see if we are searching via predicate string
+    var locatorStrategy = args[1].req.body.using;
 
-    return await super.executeCommand(...arguments);
+    if ("-ios predicate string" === locatorStrategy) {
```

Comment:
we typically do equality checks the other way around (variable first), mind switching?